client/src/pages/SwapPage/queries.ts

```diff
@@ -0,0 +1,64 @@
+import { ExchangeContractAbi } from 'src/types/contracts';
+import { ActiveInput, SwapState } from './types';
+
+const getSwapWithMaximumRequiredAmount = async (
+  contract: ExchangeContractAbi,
+  assetId: string,
+  amount: bigint
+) => {
+  const requiredAmount = await contract.callStatic.get_swap_with_maximum(amount, {
+    forward: [0, assetId],
+  });
+  return requiredAmount;
+};
+
+const getSwapWithMinimumMinAmount = async (
+  contract: ExchangeContractAbi,
+  assetId: string,
+  amount: bigint
+) => {
+  const minAmount = await contract.callStatic.get_swap_with_minimum(amount, {
+    forward: [0, assetId],
+  });
+  return minAmount;
+};
+
+export const queryPreviewAmount = async (
+  contract: ExchangeContractAbi,
+  { amount, direction, from }: SwapState
+) => {
+  if (direction === ActiveInput.to && amount) {
+    const previewAmount = await getSwapWithMaximumRequiredAmount(contract, from, amount);
+    return previewAmount;
+  } else if (amount) {
+    const previewAmount = await getSwapWithMinimumMinAmount(contract, from, amount);
+    return previewAmount;
+  }
+  return null;
+};
+
+export const swapTokens = async (
+  contract: ExchangeContractAbi,
+  { from, direction, amount }: SwapState
+) => {
+  const deadline = 1000;
```

Comment:
This could be a constant `DEADLINE` in the global scope (avoid re-use memory on each execution)